### DISCUSSION TITLE
Ulauncher Manjaro

### DISCUSSION THREAD

**Discussion by Zaiidmoumni** [Ulauncher/Ulauncher#1396] [Q&A] [answered]
Anyone has any idea why the `ctrl + space` shortcut does not work to toggle Ulauncher ? I'm using Linux  Manjaro - Gnome

**Comment by friday** [comment, marked as answer]
https://github.com/Ulauncher/Ulauncher/discussions/1347

**Comment by fidelix** [comment]
Ulauncher no longer toggles. It only activates. Very weird decision to be honest, but the author said the normal way is not coming back.

**Reply by friday** [reply]
For multiple technical reasons it's not possible to support the old toggle behavior while fixing much more important Wayland compatibility issues, rewriting Ulauncher as a standard GTK app, removing dependencies and fixing the race condition to set focus. So in order to fix all these issues in Ulauncher V6 beta it was a very simple decision to remove the toggle behavior (which also never made sense).

If you actually wanted to understand the reasoning behind the changes I'm pretty sure I explained them thoroughly in the PRs. Both the hotkey change, focus fixes (several PRs) and the main GTK app architecture PR relate to this.

Additionally, having the same key combination to show and hide the app is obviously not friendly for users with hand tremors.

We could potentially introduce more configurable key bindings in Ulauncher, which might make it possible to set Ulauncher up to behave like before, but why would anyone prefer the behavior to press two keys to close Ulauncher, when they can press one key (ESC), which is also the standard key to close modal/overlay apps and windows?

Your muscle memory isn't fixed. Just because you learned one way doesn't mean you can't unlearn and use the standard way, which is better to learn because you can use it for other apps too.

Also, nothing in the original question indicated that the user was asking about the V6 beta, or about the toggle behavior specifically. And this thread was posted and answered a year ago...

**Reply by fidelix** [reply]
> but why would anyone prefer the behavior to press two keys to close Ulauncher, when they can press one key (ESC), which is also the standard key to close modal/overlay apps and windows?

Indeed that might be the case for the author of this topic.
But it wouldn't always be the case. The normal usecase would be a single key to toggle (in fact it's possible to do that with most launchers). That is objectively less work for the majority of users.

> Your muscle memory isn't fixed. Just because you learned one way doesn't mean you can't unlearn and use the standard way, which is better to learn because you can use it for other apps too.

In this case your suggestion is the non-standard way. From my perspective, it's not me who should change, it's the app that has to be configurable. If it isn't, I simply won't use it. Which is fine, you have no obligation to make it configurable, but I'm not adapting my workflow to an objectively inferior change.

Indeed the author might not have been asking about v6, but there was no way to know. I thought it would be useful for anyone passing by. I arrived here from a search engine, I'm sure others would as well.

I did try to read the PRs and changes to understand the technical reasoning, and couldn't find a final justification for this, just you mentioning the normal behaviour would not be coming back. Perhaps I didn't search enough. I apologize if that was the case.

But I am absolutely positive it doesn't have to be this way from a technical perspective. I read the code. But like I said, you are under no obligation to change anything.

Regardless of anything, thanks for opening the source of this software. It doesn't have to work for everyone. It works for plenty of people. It's fine that it doesn't work for me.
